docarray/array/storage/elastic/find.py

```diff
@@ -53,25 +60,189 @@ def _find_similar_vectors(self, query: 'ElasticArrayType', limit=10):
+
         :param query: input supported to be stored in Elastic. This includes any from the list '[np.ndarray, tensorflow.Tensor, torch.Tensor, Sequence[float]]'
         :param limit: number of retrieved items
 
         :return: DocumentArray containing the closest documents to the query if it is a single query, otherwise a list of DocumentArrays containing
            the closest Document objects for each of the queries in `query`.
-
         """
-
-        num_rows, _ = ndarray.get_array_rows(query)
+        if isinstance(query, str):
+            search_method = functools.partial(
+                self._find_similar_documents_from_text, index=index
+            )
+            num_rows = 1
+            query = [query]
+        elif isinstance(query, list) and isinstance(query[0], str):
+            search_method = functools.partial(
+                self._find_similar_documents_from_text, index=index
+            )
+            num_rows = len(query)
+        else:
+            search_method = self._find_similar_vectors
+            query = np.array(query)
+            num_rows, n_dim = ndarray.get_array_rows(query)
+            if n_dim != 2:
+                query = query.reshape((num_rows, -1))
 
         if num_rows == 1:
-            return [self._find_similar_vectors(query[0], limit=limit)]
+            # if it is a list do query[0]
+            return [search_method(query[0], limit=limit)]
         else:
             closest_docs = []
             for q in query:
-                da = self._find_similar_vectors(q, limit=limit)
+                da = search_method(q, limit=limit)
                 closest_docs.append(da)
             return closest_docs
+
+    def find(
+        self: 'T',
+        query: Union['DocumentArray', 'Document', 'ArrayType', Dict],
+        metric: Union[
+            str, Callable[['ArrayType', 'ArrayType'], 'np.ndarray']
+        ] = 'cosine',
+        limit: Optional[Union[int, float]] = 20,
+        metric_name: Optional[str] = None,
+        exclude_self: bool = False,
+        only_id: bool = False,
+        index: str = 'text',
+        **kwargs,
+    ) -> Union['DocumentArray', List['DocumentArray']]:
+        """Returns approximate nearest neighbors given an input query.
+
+        :param query: the input query to search by
+        :param limit: the maximum number of matches, when not given defaults to 20.
+        :param metric_name: if provided, then match result will be marked with this string.
+        :param metric: the distance metric.
+        :param exclude_self: if set, Documents in results with same ``id`` as the query values will not be
+                        considered as matches. This is only applied when the input query is Document or DocumentArray.
+        :param only_id: if set, then returning matches will only contain ``id``
+        :param index: if the query is a string, text search will be performed on `index`, otherwise, ignored. By
```

Comment:
again this change should be applied to the parent find mixin, we no more want to do this whole change here
```suggestion
        :param index: if the query is a string or list of string, text search will be performed on `index`, otherwise, ignored. By
```

Reply:
I see, updated